src/main/java/org/spongepowered/server/plugin/VanillaPluginContainer.java

```diff
@@ -40,8 +42,17 @@
     VanillaPluginContainer(Injector injector, Class<?> pluginClass, PluginMetadata metadata, Optional<Path> source) {
         super(metadata, source);
 
-        this.injector = injector.createChildInjector(new PluginModule(this, pluginClass));
-        this.instance = Optional.of(this.injector.getInstance(pluginClass));
+        if (SpongeGuice.isModular(pluginClass)) {
+            try {
+                this.instance = Optional.of(pluginClass.newInstance());
+            } catch (final IllegalAccessException | InstantiationException e) {
+                throw new IllegalStateException("Could not construct instance of plugin '" + pluginClass + "'.");
```

Comment:
A common reason for a plugin reaching this would be them trying to use ctor injections while being a module right? Perhaps we should detect this to throw a better error warning them of this behaviour?